### PR TITLE
Add migrations docs 📚

### PR DESCRIPTION
**Pull request by lluia**
## Summary ❤️

I wrote _extensive_ documentation on how to migrate to `2.0.0`.

It contains two sections:
- **Overview**
- **Primer**

**Overview** contains the details of the migration but most people will refer to the **Primer** to get their apps up to date quickly 🏎

It can be previewed [here](https://github.com/zopaUK/react-components/blob/docs/v2-migration-guide/etc/docs/migration-v2.md).

( _I also updated the main `README.md` to point to it and took the opportunity to clean it up a bit_ )

### CODE REVIEW

File: etc/docs/migration-v2.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+| `<Title3 />`   | `<Heading as="h1" />`   |
+| `<Header4 />`  | `<Heading as="h4" />`   |
+| `<Lead />`     | `<Text size="lead" />`  |
+| `<SubHead />`  | `<Text as="p" />`       |
+| `<Caption />`  | `<Text size="small" />` |
+| `<HelpText />` | `<Text as="p" />`       |
+
+## Icons
+
+_Just the exported name of some icons changes for consistency_ )
```
Comment (poteirard) [inline etc/docs/migration-v2.md]: And also the default inactiveColor :)

File: etc/docs/migration-v2.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+See this article for more in depth of the [benefits of dot notation](https://medium.com/@skovy/using-component-dot-notation-with-typescript-to-create-a-set-of-components-b0b2aad4892b).
+
+Whichever component had a `color` prop, that color prop will only accept one of the new [defined colors](http://localhost:6060/#/Content?id=colors), or a subset of them where it makes sesne.
+
+## What's new ?
+
+#### [`<Text />`](https://zopauk.github.io/react-components/#/Components/Atoms/Text)
+
+Whenever you have to render short or long body text on your app, use the new `<Text />` component rather than HTML tags directly ( `<p>`, `<span`>, etc... ).
+It's built too match the latest specs on typography and give you enough flexibility to match any desing.
```
Comment (poteirard) [inline etc/docs/migration-v2.md]: typo `design`

Comment (lluia) [reply etc/docs/migration-v2.md]: and `too` 😄

File: etc/docs/migration-v2.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+#### [`<Heading />`](https://zopauk.github.io/react-components/#/Components/Atoms/Heading)
+
+Whenever you need to render a heading in your app, use the new `<Heading />` component rather than HTML tags directly ( `<h1>`, `<h2>`, etc.. ).
+
+The `as` prop is mandatory to make you think which heading tag you're rendering.
+
+🚨 &nbsp; **Please dont't render paragraph text as a heading:**
+
+```
```
Comment (lluia) [inline etc/docs/migration-v2.md]: Syntax highlighting missing `jsx`

**Pull request merged**